### PR TITLE
Templates from vcs

### PR DESCRIPTION
**Pull request by mattben**
ignore the variable mentions. Templates is what I meant.

### CODE REVIEW

File: frontend/src/js/App.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import React from 'react'
 import ReactDOM from 'react-dom'
 import AppContainer from './containers/AppContainer.jsx'
-import configureStore from './Store.js'
+import {configureStore} from './Store.js'
 import {Provider} from 'react-redux'
-
+console.log('configure store', configureStore)
```
Comment (chaosphere2112) [inline frontend/src/js/App.js]: Please remove logging code

File: frontend/src/js/reducers/Reducer.js [hunk 2]
```diff
@@ -60,13 +62,33 @@ const gmListReducer = (state = test_gms, action) => {
     }
 }
 
-const templateListReducer = (state = test_temps, action) => {
+const templateListReducer = (state = [], action) => {
+    console.log('template state', state, action);
```
Comment (chaosphere2112) [inline frontend/src/js/reducers/Reducer.js]: Please remove logging code

File: frontend/src/js/reducers/Reducer.js [hunk 3]
```diff
@@ -60,13 +62,33 @@ const gmListReducer = (state = test_gms, action) => {
...
+    if (!state.length && action.type != 'INITIALIZE_TEMPLATE_VALUES'){
+        console.log('calling getTemplates');
```
Comment (chaosphere2112) [inline frontend/src/js/reducers/Reducer.js]: Please remove logging code

File: frontend/src/js/reducers/Reducer.js [hunk 4]
```diff
@@ -60,13 +62,33 @@ const gmListReducer = (state = test_gms, action) => {
...
+        case 'INITIALIZE_TEMPLATE_VALUES':
+            return action.templates;
         default:
             return state
     }
 }
 
+const getTemplates = () => {
+    //var templates = ['one', 'two', 'there']'
+    console.log('getting templates');
```
Comment (chaosphere2112) [inline frontend/src/js/reducers/Reducer.js]: Please remove logging code

**Comment by sampsonbryce** [comment]
Harsh

File: backend/vcdat/app.py [hunk 5]
```diff
@@ -18,5 +20,11 @@ def serve_resource_file(path):
         print 'Unable to serve file ', path
         return send_from_directory(dir_path, 'taco')
 
+@app.route("/getTemplates")
+def get_templates():
+    templates = sorted(vcs.elements['template'].keys())
+    print templates
```
Comment (chaosphere2112) [inline backend/vcdat/app.py]: Please remove logging code

Comment (mattben) [inline backend/vcdat/app.py]: I removed them all from the next pull request. I also need to fix the conflicts with master.

Comment (chaosphere2112) [inline backend/vcdat/app.py]: (y)

**Comment by mattben** [comment]
busted branch

**Pull request closed without merging**